verilog/analysis/checkers/struct_union_name_style_rule.cc

```diff
@@ -79,15 +82,65 @@ void StructUnionNameStyleRule::HandleSymbol(const verible::Symbol& symbol,
       // Neither a struct nor union definition
       return;
     }
-    const auto* identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
+    const auto *identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
     const auto name = ABSL_DIE_IF_NULL(identifier_leaf)->get().text();
-    if (!verible::IsLowerSnakeCaseWithDigits(name) ||
+    // get rid of the exceptions specified by user
+    std::string name_filtered;
+    if (!exceptions_.empty()) {
+      name_filtered.reserve(name.length());
+      const auto &underscore_separated_parts = absl::StrSplit(name, '_');
+      if (name[0] != '_') {
+        for (const auto &ns : underscore_separated_parts) {
+          if (std::find(exceptions_.begin(), exceptions_.end(), ns) ==
+              exceptions_.end()) {
+            name_filtered.append(ns);
+          }
+        }
+      } else {
+        name_filtered.append("_");
+      }
+    }
+    const absl::string_view relevant_name =
+        exceptions_.empty() ? name : name_filtered;
+    if (!verible::IsLowerSnakeCaseWithDigits(relevant_name) ||
         !absl::EndsWith(name, "_t")) {
       violations_.insert(LintViolation(identifier_leaf->get(), msg, context));
     }
   }
 }
 
+absl::Status StructUnionNameStyleRule::Configure(
+    absl::string_view configuration) {
+  using verible::config::SetString;
+  std::string raw_tokens;
+  auto status = verible::ParseNameValues(
+      configuration, {{"exceptions", SetString(&raw_tokens)}});
+
+  if (!raw_tokens.empty()) {
+    const auto &exceptions = absl::StrSplit(raw_tokens, ',');
+    for (const auto &ex : exceptions) {
+      const auto &e =
+          std::find_if_not(ex.begin(), ex.end(), absl::ascii_isalnum);
+      // eliminate exceptions with invalid characters
+      if (e != ex.end()) continue;
+
+      const auto &alpha =
+          std::find_if(ex.begin(), ex.end(), absl::ascii_isalpha);
+      // find first alpha sign
+      // check if string do not start with alpha sign
+      // and if there are any alpha signs
+      if (alpha == ex.begin() || alpha == ex.end()) continue;
```

Comment:
But isn't a 'numbers followed by a particular unit' is what we'd like to offer ?
If I have a configuration `GiB`, then I'd allow things that contain `foo_12GiB_bar_t`
(#385 states _" number followed by a unit"_ which means not a specific number, just something numeric.)

So the configuration should not (only) explicitly ask for a particular number, just that the exception is something found in the string _prepended_ by something that looks like  a number.

So if someone would like to allow `B`, `GiB` and `J`, then strings such as
`foo_12B_bar_t`, `foo_13B_t`, `baz_42GiB_t` or `quux_2134J_energy_t` would be allowed.
Strings like `foo_B_t` or `baz_GiB_t` or `baz2_GiB_t` would _not_ be allowed (because they are not prepended with a digit).

The exception can be made of course more specific, so `12B` would _only_ allow `foo_12B_t` but not `foo_13B_t`